### PR TITLE
Janitorial: Disparate cleanup tasks

### PR DESCRIPTION
**Pull request by tbradsha**
This includes a variety of small cleanup tasks I ran across while scouring the codebase.

## Proposed changes:

This PR includes the following:
* Remove Phan suppressions: c15703438e, a792185799
* Remove a WP 6.5 polyfill added in #38428: 01bb8aec96
* Remove/updated `is_callable` checks: 6e6eab821b, 0dd6d173e7, a6fefc374a
* Update two missed instances of `setMethods` → `onlyMethods`: d7458f0120
* Update one instance of `private static` to `private const`: 3240990ea8
* Add type hinting: 4f834228c6, b0d86091ab
* Switched a `packages/changelogger` deprecation to a throw: 0674faf185, d143ebe0ae, 2a6feee5ef
* Removed some unneeded CSS: 3bdaac9338
* Added/updated/removed comments: 7842f7df27, 69aed20ea4, 82bfe8e277
* Removed fallbacks in CRM: 565292bd3c
* Removed unreachable logic in CI tool setup: 460e14b650
* Removed unneeded checks in CRM, Super Cache, and Vaultpress: 17b7094bd6, 5fd572037f, f1f7025d91
* Fixed changelogger error when running on macOS: e66c7aa2bd

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

Do the changes make sense? Are CI checks happy?

### CODE REVIEW

File: projects/packages/assets/changelog/remove-pre_wp6.6_and_php7.2_code [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: removed
+Comment: Additional cleanup of old PHP + WP version support.
+
+
```
Comment (anomiex) [inline projects/packages/assets/changelog/remove-pre_wp6.6_and_php7.2_code]:
```suggestion
Significance: major
Type: removed

Remove JSX runtime polyfill, now that we've dropped support for WordPress < 6.6.
```

File: projects/packages/autoloader/changelog/remove-pre_wp6.6_and_php7.2_code [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: removed
+Comment: Additional cleanup of old PHP + WP version support.
+
+
```
Comment (anomiex) [inline projects/packages/autoloader/changelog/remove-pre_wp6.6_and_php7.2_code]:
```suggestion
Significance: major
Type: removed

Remove support for WordPress 6.5 and earlier.
```

File: projects/packages/changelogger/changelog/upgrade_deprecation_to_exception
Comment (anomiex) [inline projects/packages/changelogger/changelog/upgrade_deprecation_to_exception:1]: This doesn't seem to have anything to do with the PHP 7.2 cleanup. Let's leave it for a separate PR, if we bother at all. Personally I don't really see the point to changing it from a deprecation warning to a throw before just removing it in a future major version bump.

File: projects/packages/changelogger/tests/php/tests/src/UtilsTest.php [hunk 3]
```diff
@@ -65,7 +65,7 @@ public function testRunCommand( $cmd, $options, $expectExitCode, $expectStdout,
 	 * Data provider for testRunCommand.
 	 */
 	public function provideRunCommand() {
-		$tmp = sys_get_temp_dir();
+		$tmp = realpath( sys_get_temp_dir() );
```
Comment (anomiex) [inline projects/packages/changelogger/tests/php/tests/src/UtilsTest.php:68]: This doesn't seem related to the PHP 7.2 cleanup either.

File: projects/packages/jetpack-mu-wpcom/src/features/custom-css/custom-css.php [hunk 4]
```diff
@@ -987,8 +987,7 @@ public static function update_custom_css_data( $args ) {
 		public static function editor_max_image_size( $dims, $size = 'medium', $context = null ) {
 			list( $width, $height ) = $dims;
 
-			// @phan-suppress-next-line PhanUndeclaredClassInCallable
-			if ( class_exists( 'Jetpack' ) && is_callable( 'Jetpack::get_content_width' ) && 'large' === $size && 'edit' === $context ) {
+			if ( class_exists( 'Jetpack' ) && 'large' === $size && 'edit' === $context ) {
```
Comment (anomiex) [inline projects/packages/jetpack-mu-wpcom/src/features/custom-css/custom-css.php:990]: Looks like a reasonable change, but doesn't seem to have anything to do with the PHP 7.2 change. Same for the similar stuff below.

I wonder if we even need the `class_exists( 'Jetpack' )` test? This seems to have been added in dd3b1d83735 to try fixing static analysis.

File: projects/packages/jetpack-mu-wpcom/src/features/replace-site-visibility/replace-site-visibility.php [hunk 5]
```diff
@@ -11,8 +11,8 @@
  * @return bool
  */
 function is_jetpack_connected() {
-	// @phan-suppress-next-line PhanUndeclaredClassMethod, PhanUndeclaredClassInCallable
-	return class_exists( 'Jetpack' ) && is_callable( 'Jetpack::is_connection_ready' ) && Jetpack::is_connection_ready();
+	// @phan-suppress-next-line PhanUndeclaredClassMethod
```
Comment (anomiex) [inline projects/packages/jetpack-mu-wpcom/src/features/replace-site-visibility/replace-site-visibility.php:14]: Looks like a reasonable change, but doesn't seem to have anything to do with the PHP 7.2 change.

In this case, I don't see any indication as to why the `is_callable` was added in a4dea908f50d6e5cb964abd4f3f5650b07c92d7f, although Phan may again be the reason.

The tricky part with adding a stub is that Simple has its own Jetpack class (fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Swrgcnpx%2Spynff.wrgcnpx.cuc%3Se%3Q37575q28-og) which doesn't entirely match Jetpack's, so whether we point to Jetpack's or make a stub from Simple's it'd be wrong for some environments. :sigh:

File: projects/packages/status/src/class-status.php [hunk 6]
```diff
@@ -45,8 +45,7 @@ public function is_offline_mode() {
 		/**
 		 * Filters Jetpack's offline mode.
 		 *
-		 * @see https://jetpack.com/support/development-mode/
-		 * @todo Update documentation ^^.
+		 * @see https://jetpack.com/support/offline-mode/
```
Comment (anomiex) [inline projects/packages/status/src/class-status.php:48]: Good change, but not related to old PHP or WP versions.

File: projects/plugins/crm/includes/class-oauth-handler.php [hunk 7]
```diff
@@ -3,7 +3,7 @@
  * Jetpack CRM
  * https://jetpackcrm.com
  *
- * OAuth connection handler. Requires PHP 7.3+
```
Comment (anomiex) [inline projects/plugins/crm/includes/class-oauth-handler.php:6]: Good change since CRM has long since been 7.4-minimum, but not related to the dropping of PHP 7.2.

File: projects/plugins/wpcomsh/private-site/private-site.php [hunk 8]
```diff
@@ -647,7 +647,7 @@ function preprocess_comment( $comment ) {
  * @return bool
  */
 function is_jetpack_connected() {
-	return is_callable( 'Jetpack::is_connection_ready' ) && Jetpack::is_connection_ready();
+	return class_exists( 'Jetpack' ) && Jetpack::is_connection_ready();
```
Comment (anomiex) [inline projects/plugins/wpcomsh/private-site/private-site.php:650]: Looks like a reasonable change, but doesn't seem to have anything to do with the PHP 7.2 or WP changes.

File: tools/cli/commands/changelog.js [hunk 9]
```diff
@@ -523,7 +523,7 @@ async function changelogArgs( argv ) {
 	argv = await validateProject( argv );
 	argv.success = `Command '${ argv.cmd || argv._[ 1 ] }' for ${
 		argv.project
-	} completed succesfully!`;
+	} completed successfully!`;
```
Comment (anomiex) [inline tools/cli/commands/changelog.js:526]: Good change, but doesn't seem to have anything to do with the PHP 7.2 or WP changes.

**Review by anomiex — COMMENTED** [review]
Spotted two more things to clean up:

* No need for composer 2.2 anymore with PHP 7.2 minimum. https://github.com/Automattic/jetpack/blob/a79218579963e78a24ac73e68aad2379485c879f/projects/packages/composer-plugin/composer.json#L17
* Looks like I forgot this comment in #38393. 😀 https://github.com/Automattic/jetpack/blob/a79218579963e78a24ac73e68aad2379485c879f/projects/plugins/jetpack/_inc/lib/debugger/class-jetpack-cxn-test-base.php#L514

Also, since the title of this PR is explicitly about the WP 6.6 / PHP 7.2 change, I flagged a bunch of stuff that isn't related to that. I'd be satisfied if the title and description are changed to encompass general cleanups instead of being so specific.

File: projects/packages/changelogger/changelog/upgrade_deprecation_to_exception
Comment (tbradsha) [reply projects/packages/changelogger/changelog/upgrade_deprecation_to_exception:1]: The reason I did this is because it has a major version bump forthcoming, and had this comment:

```
// @todo Sometime for a major version bump, have this throw instead. Then remove it for a major bump after that.
```

I'll defer to your final judgment. :^)

File: projects/packages/changelogger/tests/php/tests/src/UtilsTest.php (hunk 3 above)
Comment (tbradsha) [reply projects/packages/changelogger/tests/php/tests/src/UtilsTest.php:68]: Changed PR title.

File: projects/packages/jetpack-mu-wpcom/src/features/custom-css/custom-css.php (hunk 4 above)
Comment (tbradsha) [reply projects/packages/jetpack-mu-wpcom/src/features/custom-css/custom-css.php:990]: Changed PR title.

As for whether we need it, I wasn't confident enough in the load order and knowledge of the environment to know if there's an edge case where the class might not exist. 😓

File: projects/packages/jetpack-mu-wpcom/src/features/replace-site-visibility/replace-site-visibility.php (hunk 5 above)
Comment (tbradsha) [reply projects/packages/jetpack-mu-wpcom/src/features/replace-site-visibility/replace-site-visibility.php:14]: Updated the PR title. Good to know about the Simple stub! At least in this case it still should be safe.

File: projects/packages/status/src/class-status.php (hunk 6 above)
Comment (tbradsha) [reply projects/packages/status/src/class-status.php:48]: Updated PR title.

File: projects/plugins/crm/includes/class-oauth-handler.php (hunk 7 above)
Comment (tbradsha) [reply projects/plugins/crm/includes/class-oauth-handler.php:6]: Updated PR title.

File: projects/plugins/wpcomsh/private-site/private-site.php (hunk 8 above)
Comment (tbradsha) [reply projects/plugins/wpcomsh/private-site/private-site.php:650]: Updated PR title.

File: tools/cli/commands/changelog.js (hunk 9 above)
Comment (tbradsha) [reply tools/cli/commands/changelog.js:526]: Updated PR title.

**Comment by tbradsha** [comment]
> No need for composer 2.2 anymore with PHP 7.2 minimum.
 
Indeed, and the same goes with `symfony/console` and `symfony/process` (and probably others). I'm going to defer that to a separate PR, and I'm also not sure the best way to do it... 😅

> Looks like I forgot this comment

82bfe8e277

> I'd be satisfied if the title and description are changed to encompass general cleanups

Adjusted.

**Review by anomiex — APPROVED** [review]
Looks reasonable. Haven't tested though.
